src/js_results.hpp

```diff
@@ -314,9 +314,6 @@ void ResultsClass<T>::index_of(ContextType ctx, Fn& fn, Arguments& args, ReturnV
     try {
         ndx = fn(args[0]);
     }
-    catch (realm::Results::IncorrectTableException&) {
-        throw std::runtime_error("Object type does not match the type contained in result");
-    }
```

Comment:
I removed this because now Core throws an error with a much better error message and we don't need the translation here.